File: examples/ucv_test_publisher/scripts/robot_publisher.m
function robot_publisher()
    % initialize the ros node
    rosinit('localhost');

    % create publisher which can talk to Robot and tell it to move
    pub = rospublisher('/cmd_vel', 'geometry_msgs/Twist');

    % create a twist message and add linear x and angular z values
    move_cmd = rosmessage(pub);

    % set publish rate at 10 Hz
    rate = rosrate(10);

    % -- move straight --
    disp('Moving straight');
    move_cmd.Linear.X = 0.3;
    move_cm.Angular.Z = 0.0;

    % for the next 3 seconds publish cmd_vel move commands
    now = rostime('now');
    while rostime('now') - now < rosduration(3)
        send(pub, move_cmd);
        waitfor(rate);
    end

    % -- rotating counterclockwise --
    disp('Rotating');
    move_cmd.Linear.X = 0.0;
    move_cmd.Angular.Z = 0.2;

    % for the next 3 seconds publish cmd_vel move commands
    now = rostime('now');
    while rostime('now') - now < rosduration(3)
        send(pub, move_cmd);
        waitfor(rate);
    end

    % -- stop --
    disp('Stopping');
    move_cmd.Linear.X = 0.0;
    move_cmd.Angular.Z = 0.0;

    now = rostime('now');
    while rostime('now') - now < rosduration(1)
        send(pub, move_cmd);
        waitfor(rate); 
    end

    disp('Exit');

    % shutdown ROS node
    rosshutdown;
end

File: examples/ucv_test_publisher/scripts/robot_publisher.py
#!/usr/bin/env python3
from geometry_msgs.msg import Twist

import rospy
import time

def move_robot():
    rospy.init_node('robot_publisher', anonymous=True)

    # create a publisher which can talk to Robot and tell it to move
    pub = rospy.Publisher('/cmd_vel', Twist, queue_size=10)

    # set publish rate at 10 Hz
    rate = rospy.Rate(10)

    # create a Twist message and add linear x and angular z values
    move_cmd = Twist()

    # --- move straight ---
    print('Moving straight')
    # move in X axis at 0.3 m/s
    move_cmd.linear.x = 0.3
    move_cmd.angular.z = 0.0

    now = time.time()
    # for the next 3 seconds publish cmd_vel move commands
    while time.time() - now < 3:
        pub.publish(move_cmd)
        rate.sleep()

    # --- rotating counterclockwise ---
    print('Rotating')
    move_cmd.linear.x = 0.0
    # rotate at 0.2 rad/sec
    move_cmd.angular.z = 0.2

    now = time.time()
    # for the next 3 seconds publish cmd_vel move commands
    while time.time() - now < 3:
        pub.publish(move_cmd)
        rate.sleep()

    # --- stopping ---
    print('Stopping')
    move_cmd.linear.x = 0.0
    move_cmd.angular.z = 0.0

    now = time.time()
    while time.time() - now < 1:
        pub.publish(move_cmd)
        rate.sleep()

    print('Exit')

if __name__ == '__main__':
    try:
        move_robot()
    except rospy.ROSInterruptException:
        pass


